eks-hybrid-ssm-setup.sh

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env bash
+
+exec >&2
+set -eux -o pipefail
+
+HOST_ROOTFS="/.bottlerocket/rootfs"
+SSM_AGENT_PERSISTENT_STATE_DIR="${HOST_ROOTFS}/local/host-containers/control/ssm"
+SSM_AGENT_REGISTRATION="${SSM_AGENT_PERSISTENT_STATE_DIR}/registration"
+mkdir -p "${SSM_AGENT_PERSISTENT_STATE_DIR}"
+ENABLE_CREDENTIALS_FILE="false"
+DRY_RUN="false"
+SSM_ACTIVATION_ID_REGEX="^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$"
+
+set +x
+for opt in "$@"; do
+    optarg="$(expr "${opt}" : '[^ =]*[= ]\(.*\)')"
+    case "${opt}" in
+       --region=*) AWS_REGION="${optarg}" ;;
+       --activation-code=*) SSM_ACTIVATION_CODE="${optarg}" ;;
+       --activation-id=*) SSM_ACTIVATION_ID="${optarg}" ;;
+       --enable-credentials-file=*) ENABLE_CREDENTIALS_FILE="${optarg}" ;;
+        --dry-run=*) DRY_RUN="${optarg}" ;;
+    esac
+done
+set -x
+
+validate_activation_id() {
+    if ! [[ $SSM_ACTIVATION_ID =~ ^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$ ]]; then
+        echo "Error: Invalid activation ID format" >&2
+        exit 1
+    fi
+}
+
+extract_ssm_agent() {
+  pushd / >/dev/null 2>&1
+  dnf -yq install 'dnf-command(download)' cpio rsync
+  dnf download amazon-ssm-agent
+  rpm2cpio amazon-ssm-agent-*.rpm | cpio -idmu '*bin/amazon-ssm-agent'
+  popd >/dev/null 2>&1
+}
+
+register_hybrid_node() {
+  amazon-ssm-agent \
+    -register \
+    -region "${AWS_REGION}" \
+    -code "${SSM_ACTIVATION_CODE}" \
+    -id "${SSM_ACTIVATION_ID}" \
+    -disableSimilarityCheck
+}
+
+persist_ssm_state() {
+  rsync -aq \
+    /var/lib/amazon/ssm/ \
+    "${SSM_AGENT_PERSISTENT_STATE_DIR}"
+}
+
+set_hostname_settings() {
+  local ssm_node_id
+  local cluster_name
+  local region
+  ssm_node_id="$(jq -r '.ManagedInstanceID' "${SSM_AGENT_PERSISTENT_STATE_DIR}/registration")"
+  cluster_name="$(apiclient get settings.kubernetes.cluster-name | jq -r ".settings.kubernetes.\"cluster-name\"")"
+  region="$(apiclient get settings.aws.region | jq -r ".settings.aws.region")"
+  variant_id="$(apiclient get os.variant_id | jq -r ".os.\"variant_id\"")"
+  version_id="$(apiclient get os.version_id | jq -r ".os.\"version_id\"")"
+
+  apiclient set \
+    network.hostname="${ssm_node_id}" \
+    kubernetes.hostname-override="${ssm_node_id}" \
+    kubernetes.provider-id="eks-hybrid:///$region/$cluster_name/$ssm_node_id" \
+    "settings.kubernetes.node-labels.\"os.bottlerocket.aws/variant\""="${variant_id}" \
+    "settings.kubernetes.node-labels.\"os.bottlerocket.aws/version\""="${version_id}"
+}
+
+symlink_aws_creds() {
+  local control_creds
+  local host_creds
+  control_aws_dir="${HOST_ROOTFS}/run/host-containerd/io.containerd.runtime.v2.task/default/control/rootfs/root/.aws"
+  control_aws_dir_relative_to_host="/run/host-containerd/io.containerd.runtime.v2.task/default/control/rootfs/root/.aws"
```

Comment:
We should declare all these function variables as local, instead of only some of them. (Same for the other function variables, like `variant_id` in `set_hostname_settings`.)